Modix.Services/UserInfo/UserInfoCommand.cs

```diff
@@ -0,0 +1,297 @@
+
+            var usersAbleToRemove = new[]
+            {
+                discordUser,
+                request.Message.Author,
+            };
+
+            await _autoRemoveMessageService.RegisterRemovableMessageAsync(
+                usersAbleToRemove,
+                embedBuilder,
+                async (builder) => await request.Message.ReplyAsync(string.Empty, builder.Build()));
+        }
+
+        private async Task<UserInfoUserDto?> GetUserAsync(ulong userId, ulong guildId, CancellationToken cancellationToken)
+        {
+            return await _modixContext
+                .GuildUsers
+                .WhereUserInGuild(userId, guildId)
+                .Select(x => new UserInfoUserDto
+                {
+                    FirstSeen = x.FirstSeen,
+                    LastSeen = x.LastSeen,
+                    Infractions = x.Infractions
+                        .AsQueryable()
+                        .Where(InfractionQueryExtensions.IsActive())
+                        .Select(d => new UserInfoUserDto.UserInfoInfractionDto
+                        {
+                            Type = d.Type,
+                            Created = d.CreateAction.Created,
+                            Reason = d.Reason,
+                        }).ToList(),
+                    Promotions = x.PromotionCampaigns
+                        .AsQueryable()
+                        .Where(PromotionCampaignQueryExtensions.IsAccepted())
+                        .Select(d => new UserInfoUserDto.UserInfoPromotionCampaignDto
+                        {
+                            ClosedDate = d.CloseAction!.Created,
+                            TargetRoleId = d.TargetRoleId,
+                        }).ToList(),
+                })
+                .FirstOrDefaultAsync(cancellationToken);
+        }
+
+        private async Task<UserInfoParticipationDto?> GetParticipationStatisticsAsync(IGuild guild, ulong userId, CancellationToken cancellationToken)
+        {
+            const int NumberOfDaysInPeriodToCount = 30;
+            var now = DateTimeOffset.UtcNow;
+            var startingThreshold = now.AddDays(-NumberOfDaysInPeriodToCount);
+            var weekThreshold = now.AddDays(-7);
+
+            var messages = await _modixContext
+                .Messages
+                .WhereIsFromGuildUser(userId, guild.Id)
+                .WhereCountsTowardsParticipation()
+                .Where(x => x.Timestamp > startingThreshold)
+                .GroupBy(x => new { x.Timestamp.ToUniversalTime().Date, x.ChannelId })
+                .Select(x => new
+                {
+                    Date = x.Key.Date,
+                    ChannelId = x.Key.ChannelId,
+                    NumberOfMessages = x.Count(),
+                }).ToListAsync(cancellationToken: cancellationToken);
+
+            if (!messages.Any())
+            {
+                return null;
+            }
+
+            var rankedUsers = await _modixContext
```

Comment:
Why query for all the rankings, instead of filtering down to just the user we care about, and letting the DB attach a row number to use as the "rank" value?

Reply:
I could put it into one, but my methodology at this point is smaller, but more frequent payloads. Grouping by user as well would bring back a larger payload... I think? Can be changed during testing.

Reply:
No, I'm not referring to merging this together with other queries, I means specficially only pulling back one row with this query, not all rows. You can select the "index" or "rownum" of a row in SQL, then have an outer WHERE clause to limit to just the user we want. Not entirely sure if this is something we can do with EF, but I would bet it is.